plugins/modules/podman_container.py

```diff
@@ -721,6 +724,16 @@
       - Publish all exposed ports to random ports on the host interfaces. The
         default is false.
     type: bool
+  quadlet_file_path:
```

Comment:
Should we allow the user to input any dir where quadlet unit files are created?
I understand that, as one of the systemd generators, quadlet will scan [the specific directories](https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#podman-unit-search-path) during boot and when `systemctl daemon-reload`. Podman has the env var `QUADLET_UNIT_DIRS` but this value seems to be [mainly used for CI tests](https://github.com/containers/podman/blob/v4.9.3/cmd/quadlet/main.go#L107-L113). So it might be better to make only these directories configurable, let's say /etc/containers/systemd/ for rootful and $HOME/.config/containers/systemd/
 for rootless.

Reply:
I think it's worth to have some default folder, but also to allow to choose one. Since we have multiple possible folders for that, for example 3 for rootless:
```
$XDG_CONFIG_HOME/containers/systemd/ or ~/.config/containers/systemd/
/etc/containers/systemd/users/$(UID)
/etc/containers/systemd/user
```
But I agree we can take one of them as a default and not to require `quadlet_file_path` to be set.

Reply:
Actually we still need to have a file name, so probably we'll need this variable to be set anyway.

Reply:
I've been testing this for the last few weeks and I have some questions about what you're talking about here:

# Preamble

According to the official documentation, as @nishipy said, there is only a [limited set of directories](https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#podman-rootful-unit-search-path) in which systemd will look for Quadlet files:

- For rootfull: `/etc/containers/systemd` and `/usr/share/containers/systemd`
- For rootless: `$XDG_CONFIG_HOME/containers/systemd`, `~/.config/containers/systemd/`, `/etc/containers/systemd/users/$(UID)` and `/etc/containers/systemd/users`

Also, as specified by the official documentation under [Container units [Container]](https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#container-units-container):

> Container units are named with a `.container` extension and contain a `[Container]` section describing the container that is run as a service. [...] However, some options also affect the details of how systemd is set up to run and interact with the container.
>
> By default, the Podman container has the same name as the unit, but with a `systemd-` prefix, i.e. a `$name.container` file creates a `$name.service` unit and a `systemd-$name` Podman container. The `ContainerName` option allows for overriding this default name with a user-provided one.

As the [name of the container is required](https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_container_module.html#parameter-name) in Ansible's `containers.podman.podman_container` module, the default behaivour of Quadlets is not reacheable.

# Propousal

* Add a `quadlet_basepath` property which allow users select in which path to store the Quadlet files. It could have a default value for rootfull and rootless.
* Use the `name` property for file naming, so the filename is `<name>.container`.
* Delete `quadlet_file_path` property as it would be generated: `<quadlet_basepath>/<name>.container`.
* Add a `container_name` property which allow users override default behaivour for container naming.


*NOTE:* this also applies to other unit types (network, volume, etc)

Reply:
@alorle yeah, all this I implement in the new version, came to same conclusions, thanks!